torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -12,6 +12,11 @@
 #include <c10/util/Exception.h>
 #include <c10/util/Optional.h>
 
+#if defined(NCCL_MAJOR) && (NCCL_MAJOR == 2) && defined(NCCL_MINOR) && \
+    (NCCL_MINOR >= 14)
+#define NCCL_HAS_COMM_NONBLOCKING
```

Comment:
Not sure why this needs to be redefined here in order to work when a definition already exists in `ProcessGroupNCCL.hpp`.